src/printer.js

```diff
@@ -2934,6 +3035,26 @@ function isEmptyBlock(doc) {
   return str === "{}";
 }
 
+
+function shouldTypeScriptTypeAvoidColon(path) {
+  // As the special TS nodes isn't returned by the node helpers,
+  // we use the stack directly to get the parent node.
+  const parent = path.stack[path.stack.length - 3]
```

Comment:
By adding TS support to ast-types the usual `.getParentNode()` should work so hopefully that will be gone soon.